app/services/email.py

```
@@ -0,0 +1,162 @@
+                EmailRecord.caller_id == caller_id,
+                EmailRecord.idempotency_key == idempotency_key
+            )
+        )
+        return result.scalar_one_or_none()
+    
+    async def update_status(
+        self,
+        db: AsyncSession,
+        email_id: str,
+        status: EmailStatus,
+        error_message: str = None,
```

Ruff (0.14.10)

PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)